### PR TITLE
Ma/permissions

### PR DESCRIPTION
**Pull request by markusahlstrand**
The auth middleware used scopes instead of permissions. This is the way it should work in all projects nowadays.

### CODE REVIEW

File: src/authentication.ts [hunk 1]
```diff
@@ -115,14 +115,14 @@ async function getJwks(env: Env, securitySchemeName: SecuritySchemeName) {
   return jwksUrls[jwksUrl];
 }
 
-function isValidScopes(token: TokenData, scopes: string[]) {
-  if (!scopes.length) {
+function isValidScopes(token: TokenData, permissions: string[]) {
```
Comment (danielgent) [inline src/authentication.ts:118]:
```suggestion
function isValidPermissions(token: TokenData, permissions: string[]) {
```

Should we rename the function?


The mismatch of plurality between `is` and `scopes` always bothered me but it's not Polish :smile:

File: src/authentication.ts [hunk 2]
```diff
@@ -161,7 +161,7 @@ export async function getUser(
   ctx: Context<Env>,
   securitySchemeName: SecuritySchemeName,
   bearer: string,
-  scopes: string[],
+  permissios: string[],
```
Comment (danielgent) [inline src/authentication.ts:164]: typo - this is probably Portuguese :smile: 
```suggestion
  permissions: string[],
```

Comment (markusahlstrand) [reply src/authentication.ts:164]: ohh.. slopppy

File: src/authentication.ts [hunk 3]
```diff
@@ -259,7 +259,7 @@ export function authenticationHandler(
       ? SecuritySchemeName.oauth2
       : SecuritySchemeName.oauth2managementApi;
 
-  const [scope] = authProvider[securitySchemeName];
+  const [permissionString] = authProvider[securitySchemeName];
```
Comment (danielgent) [inline src/authentication.ts:262]: nice rename. `scope` vs `scopes` is a recipe for disaster :smile: 

If often wonder if we should adapt hungarian typing e.g. `strPermission` to have some standardization in these use cases where we're throwing around the same thing in different types :shrug: 

Functional people would probably say everything should be a function (Martin Fowler doesn't even assign variables a lot of the times, he sees this as premature optimisation)

File: src/authentication.ts [hunk 4]
```diff
@@ -172,7 +172,7 @@ export async function getUser(
     throw new ExpiredTokenError();
   }
 
-  if (!isValidScopes(token, scopes)) {
+  if (!isValidScopes(token, permissios)) {
```
Comment (danielgent) [inline src/authentication.ts:175]: @markusahlstrand it's going to blow up here

```suggestion
  if (!isValidScopes(token, permissions)) {
```

File: src/authentication.ts (hunk 1 above)
Comment (danielgent) [reply src/authentication.ts:118]: Also of course commiting this will blow everything up

**Comment by markusahlstrand** [comment]
Did some fixes for suggested updates :)

**Pull request merged**